javaslang/src-gen/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,437 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+/*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*\
+   G E N E R A T O R   C R A F T E D
+\*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*/
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import javaslang.collection.List;
+import javaslang.concurrent.Future;
+
+import javaslang.Function3;
+import javaslang.Function4;
+import javaslang.Function5;
+import javaslang.Function6;
+import javaslang.Function7;
+import javaslang.Function8;
+import javaslang.control.Option;
+import javaslang.control.Try;
+import javaslang.concurrent.Future;
+
+/**
+ * Applicative helpers
+ *
+ * @author Daniel Dietrich
+ * @since 2.1.0
+ */
+public final class Applicative {
```

Comment:
actually I don't think so. What I do here is what `liftA2` does in haskell. The difference between applicative and monadic is more about whether a step in computation can reuse the results of previous steps in the computation. Since here the final step does NOT reuse intermediary results we do not need a monad, an applicative is enough. To give another perspective: if you have a monadic structure, you MUST perform the steps in a certain order, because a step may depend on the previous steps. If you have an applicative structure, the order is irrelevant. See also my comments in the bug: also scalaz and cats put this in the `Apply`/`Applicative` basket. I also gave an example with `liftA2` in the bug the "A" in "liftA" is for "Applicative".